code/modules/projectiles/guns/energy.dm

```diff
@@ -12,6 +12,12 @@
 	var/can_charge = 1 //Can it be charged in a recharger?
 	ammo_x_offset = 2
 	var/shaded_charge = 0 //if this gun uses a stateful charge bar for more detail
+	var/setting = -1 //Don't touch this.
```

Comment:
Its what the gun uses to initialise, used to use a separate var but carbon did not like it.